### PR TITLE
generate defmt::Format for enums

### PR DESCRIPTION
**Pull request by burrbull**
depends on #769

### CODE REVIEW

**Review by therealprof — DISMISSED** [review]
LGTM, thanks!

**Comment by Emilgardis** [comment]
this depends on #769

**Comment by burrbull** [comment]
rebased

File: src/main.rs [hunk 1]
```diff
@@ -127,7 +127,14 @@ fn run() -> Result<()> {
...
             Arg::new("impl_debug_feature")
                 .long("impl_debug_feature")
-                .help("add feature gating for block and register debug implementation")
+                .help("Add feature gating for block and register debug implementation")
+                .action(ArgAction::Set)
+                .value_name("FEATURE"),
+        )
+        .arg(
+            Arg::new("impl_defmt")
+                .long("impl-defmt")
```
Comment (Emilgardis) [inline src/main.rs:136]: this is not consistent with the other flags, unless we also bring in #771

Comment (burrbull) [reply src/main.rs:136]: do you suggest to support both `-` and `_` for all flags, or only for old?

Comment (Emilgardis) [reply src/main.rs:136]: If we go with #771, I think it makes sense to only alias snake case for old flags.
Thus, if we do include #771 then an alias on `impl-defmt` isn't needed, but maybe it's nice to include for familiarity.

Comment (burrbull) [reply src/main.rs:136]: added alias in last commit

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**